component/azstorage/block_blob.go

```diff
@@ -1698,3 +1700,29 @@ func (bb *BlockBlob) SetFilter(filter string) error {
 	bb.Config.filter = &blobfilter.BlobFilter{}
 	return bb.Config.filter.Configure(filter)
 }
+
+// SetMetadata : Set metadata property of the blob
+func (bb *BlockBlob) SetMetadata(filePath string, newMetadata map[string]*string, etag *azcore.ETag, overwrite bool) (err error) {
+	log.Trace("BlockBlob::SetMetadata : name %s", filePath)
+
+	if !overwrite {
+		attr, err := bb.GetAttr(filePath)
+		if err != nil {
+			log.Err("BlockBlob::SetMetadata : Failed to get attributes of file %s [%s]", filePath, err.Error())
+			return err
+		}
+		maps.Copy(attr.Metadata, newMetadata)
+		newMetadata = attr.Metadata
+	}
+
+	blobClient := bb.Container.NewBlockBlobClient(filePath)
+	// Set the metadata
+	_, err = blobClient.SetMetadata(context.Background(), newMetadata, &blob.SetMetadataOptions{
+		AccessConditions: &blob.AccessConditions{
+			ModifiedAccessConditions: &blob.ModifiedAccessConditions{
+				IfMatch: etag,
+			},
+		},
+	})
+	return err
```

Comment:
We need to convert the err to storage error and then handle 404 and other errors just like all other apis.